lightningd/channel/channel.c

```diff
@@ -254,6 +260,217 @@ static void handle_peer_add_htlc(struct peer *peer, const u8 *msg)
 			    "Bad peer_add_htlc: %u", add_err);
 }
 
+static void send_commit(struct peer *peer)
+{
+	tal_t *tmpctx = tal_tmpctx(peer);
+	u8 *msg;
+	secp256k1_ecdsa_signature commit_sig, *htlc_sigs;
+	size_t i;
+	struct bitcoin_tx **txs;
+	const u8 **wscripts;
+	const struct htlc **htlc_map;
+	struct pubkey localkey;
+	struct privkey local_secretkey;
+
+	/* Timer has expired. */
+	peer->commit_timer = NULL;
+
+	/* BOLT #2:
+	 *
+	 * A node MUST NOT send a `commitment_signed` message which does not
+	 * include any updates.
+	 */
+	if (!channel_sent_commit(peer->channel)) {
+		tal_free(tmpctx);
+		return;
+	}
+
+	if (!derive_simple_privkey(&peer->our_secrets.payment_basepoint_secret,
+				   &peer->channel->basepoints[LOCAL].payment,
+				   &peer->next_per_commit[REMOTE],
+				   &local_secretkey))
+		status_failed(WIRE_CHANNEL_CRYPTO_FAILED,
+			      "Deriving local_secretkey");
+
+	if (!derive_simple_key(&peer->channel->basepoints[LOCAL].payment,
+			       &peer->next_per_commit[REMOTE],
+			       &localkey))
+		status_failed(WIRE_CHANNEL_CRYPTO_FAILED,
+			      "Deriving localkey");
+
+	{ /* FIXME: extra test that keys are correct. */
```

Comment:
Because it was a debugging hack when I got it wrong.  I will simply remove it now.